### PR TITLE
Fixup Ban org.ow2.asm:asm in non-deployment modules 

### LINKED ISSUE #493: Ban org.ow2.asm:asm in non-deployment modules
Before https://github.com/quarkiverse/quarkus-cxf/pull/487, we were banning `org.ow2.asm:asm` in some modules. We should investigate whether we should/can ban it after https://github.com/quarkiverse/quarkus-cxf/pull/487 and why.

When banned, it would have to be excluded from the follwing BOM entries:

```
[ERROR]     io.quarkiverse.cxf:quarkus-cxf:1.5.0-SNAPSHOT:jar pulls banned dependencies [org.ow2.asm:asm]
[ERROR]     io.quarkiverse.cxf:quarkus-cxf-rt-features-logging:1.5.0-SNAPSHOT:jar pulls banned dependencies [org.ow2.asm:asm]
[ERROR]     io.quarkiverse.cxf:quarkus-cxf-rt-features-metrics:1.5.0-SNAPSHOT:jar pulls banned dependencies [org.ow2.asm:asm]
[ERROR]     io.quarkiverse.cxf:quarkus-cxf-rt-transports-http-hc5:1.5.0-SNAPSHOT:jar pulls banned dependencies [org.ow2.asm:asm]
[ERROR]     org.apache.cxf:cxf-rt-ws-security:3.5.3:jar pulls banned dependencies [org.ow2.asm:asm]
[ERROR]     org.apache.cxf:cxf-rt-ws-mex:3.5.3:jar pulls banned dependencies [org.ow2.asm:asm]
[ERROR]     io.quarkiverse.cxf:quarkus-cxf-rt-ws-security:1.5.0-SNAPSHOT:jar pulls banned dependencies [org.ow2.asm:asm]
```

As a first step, we can do the above exclusions and check whether some tests fail.

### PR DESCRIPTION
**Pull request by ppalaga**
Fix #493

### CODE REVIEW

File: extensions/pom.xml [hunk 1]
```diff
@@ -21,52 +21,40 @@
...
-            </plugin>
-        </plugins>
-    </build>
-
+    <profiles>
+        <profile>
+            <id>non-deployment-bans</id>
+            <activation>
+                <file>
+                    <exists>${project.basedir}/src/main/resources/META-INF/quarkus-extension.yaml</exists>
```
Comment (famod) [inline extensions/pom.xml:29]:
```suggestion
                    <exists>${basedir}/src/main/resources/META-INF/quarkus-extension.yaml</exists>
```

That placeholder rang a bell and indeed there's a warning:
```
Warning:  
Warning:  Some problems were encountered while building the effective model for io.quarkiverse.cxf:quarkus-cxf-extensions:pom:1.5.0-SNAPSHOT
Warning:  'profiles.profile[non-deployment-bans].activation.file.exists' Failed to interpolate file location ${project.basedir}/src/main/resources/META-INF/quarkus-extension.yaml for profile non-deployment-bans: ${project.basedir} expression not supported during profile activation, use ${basedir} instead
```

Ran into this myself multiple times.

**Review by famod — CHANGES_REQUESTED** [review]
LGTM, exept that property warning.

File: extensions/pom.xml (hunk 1 above)
Comment (ppalaga) [reply extensions/pom.xml:29]: Hm... I knew that profile activation is quite special, but I have not seen this before. Thanks for the hint! I used to use `${basedir}` util I saw that it is going to get removed in Maven 4: https://github.com/apache/maven/pull/696

But OK, let's go back to `${basedir}` for now.

**Comment by ppalaga** [comment]
ebcdfde: Adressed the feedback of @famod

File: extensions/pom.xml (hunk 1 above)
Comment (famod) [reply extensions/pom.xml]: Yeah, it's a nasty and confusing pitfall.

**Review by famod — APPROVED** [review]

**Pull request merged**